privateSBOMExchange/src/petra/lib/models/__init__.py

```diff
@@ -0,0 +1,63 @@
+from smt.tree import TreeMapStore, TreeMemoryStore
+from smt.tree import SparseMerkleTree
+from smt.utils import DEFAULTVALUE, PLACEHOLDER
+from smt.proof import verify_proof
+from smt.tree import SparseMerkleTree
+
+#represent SBOM in file as Merkle tree
+def SBOM_as_tree(flatten_SBOM_data,sbom_file_encoding):
+    tree = SparseMerkleTree(store=TreeMemoryStore())
+    tree_name=""
+    count=0
+    # add each SBOM field to the tree
+    for field_name,value in flatten_SBOM_data.items():
+        count+=1
+        SBOMField={field_name:value}
+        if field_name =="name":
+            tree_name=value
+        assert DEFAULTVALUE == tree.get(b"")
+        for item in SBOMField:
+            if isinstance(SBOMField[item], bool):
+                SBOMField[item]=str(SBOMField[item])
+            root1 = tree.update(item.encode(sbom_file_encoding), SBOMField[item].encode(sbom_file_encoding))
```

Comment:
What is this line doing? Why are we encoding the contents of the tree based on the SBOM file encoding? I think we should be operating on raw bytes.